ngromeconf/ngrome-website
# feat: workshop details page + manfred steyer 2024 workshop #75

#### Changes Made:

1. **Workshop Detail Page:**
   - Created a new workshop detail page for enhanced user experience.
   - Implemented a detailed structure for workshop details, including title, description, date, author information, location details, image, link, and additional custom attributes.

2. **Workshop Model Updates:**
   - Modified the workshop model to include a more detailed structure, enhancing the workshop attributes for better representation and utilization.

   ```typescript
   export interface WorkshopAttributes {
     slug: string;
     tag: string;
     title: string;
     description: string;
     date: string;
     author: {
       name: string;
       biography: string;
       image: string;
       link: string;
     };
     location: {
       name: string;
       mapsLink: string;
     };
     image: string;
     link: string;
     col: number;
     [key: string]: string | number | boolean | Date | undefined | Object;
   }
   ```

3. **Sorting and Filtering:**
   - Implemented sorting and filtering(just internally, not for user) of past workshops based on the workshop date.
   - Ensured that only workshops with dates prior to the current date are considered as past workshops.

4. **Query Params for Workshop Detail:**
   - Added query parameters to retrieve the correct workshop details by slug using the internal API.
   - Improved the retrieval process for workshop details on the workshop detail page.

5. **Additional Workshop:**
   - Added a new workshop for Manfred Steyer on 26 June.

#### How to Test:

1. Navigate to the workshops page.
2. Check the sorting and filtering of past workshops based on the workshop date.
3. Test the query parameter functionality by accessing specific workshop details clicking/using the slug.
4. Verify that the workshop details are displayed correctly with the new structure.

#### Additional Notes:

I need a bit of help to connect the tito ticket functionality and a review of the layout to understand if some changes are needed to be more inline with the current design.
